cpp/src/arrow/util/bit_stream_utils.h

```diff
@@ -190,6 +195,14 @@ class BitReader {
   /// Maximum byte length of a vlq encoded int64
   static constexpr int kMaxVlqByteLengthForInt64 = 10;
 
+  const uint8_t* GetBuffer() { return buffer_ - 1; }
+
+  int GetBufferLen() { return max_bytes_ + 1; }
+
+  void updateValOffset(int batch_size) { val_offset_ += batch_size; }
+
+  int GetValOffset() { return val_offset_; }
```

Comment:
I don't think that exporting internal buffer and changing offset from outside is a good API.
This is a reader class. So this is responsible to manage them. This API just uses this as a data container.

Can we choose another approach? For example, can we use QPL in this?

Reply:
It makes sense. I have add GetBatchWithQpl in bit_stream_utils.h and not expose the variables in BitReader. Please take a look.